pkg/actuators/machine/instances.go

```diff
@@ -450,51 +459,45 @@ func buildDescribeSecurityGroupsTag(tags []alibabacloudproviderv1.Tag) *[]ecs.De
-		vSwitchID = machineProviderConfig.VSwitchID
+	if machineProviderConfig.VSwitch == nil {
+		return "", errors.New("no vswitch configuration provided")
 	}
 
-	if machineProviderConfig.VSwitch != nil {
-		if machineProviderConfig.VSwitch.ID != "" {
-			vSwitchID = machineProviderConfig.VSwitch.ID
-		} else {
-			if machineProviderConfig.VSwitch.Tags != nil {
-				describeVSwitchesRequest := vpc.CreateDescribeVSwitchesRequest()
-				describeVSwitchesRequest.Scheme = "https"
-
-				describeVSwitchesRequest.RegionId = machineProviderConfig.RegionID
-				if machineProviderConfig.VpcID != "" {
-					describeVSwitchesRequest.VpcId = machineProviderConfig.VpcID
-				}
-				describeVSwitchesRequest.Tag = buildDescribeVSwitchesTag(machineProviderConfig.VSwitch.Tags)
+	if machineProviderConfig.VSwitch.ID != "" {
+		return machineProviderConfig.VSwitch.ID, nil
+	}
 
-				describeVSwitchesResponse, err := client.DescribeVSwitches(describeVSwitchesRequest)
-				if err != nil {
-					metrics.RegisterFailedInstanceCreate(&metrics.MachineLabels{
-						Name:      machine.Name,
-						Namespace: machine.Namespace,
-						Reason:    err.Error(),
-					})
-					klog.Errorf("error describing vswitches: %v", err)
-					return "", fmt.Errorf("error describing vswitches: %v", err)
-				}
+	if machineProviderConfig.VSwitch.Tags != nil {
+		return getVSwitchIDFromTags(machine, machineProviderConfig, client)
+	}
 
-				if len(describeVSwitchesResponse.VSwitches.VSwitch) < 1 {
-					klog.Errorf("no vswitches for given tags not found")
-					return "", fmt.Errorf("no vswitches for given tags not found")
-				}
+	return "", fmt.Errorf("no vSwitch found from configuration")
+}
 
-				vSwitchID = describeVSwitchesResponse.VSwitches.VSwitch[0].VSwitchId
-			}
-		}
+func getVSwitchIDFromTags(machine runtimeclient.ObjectKey, mpc *alibabacloudproviderv1.AlibabaCloudMachineProviderConfig, client alibabacloudClient.Client) (string, error) {
+	// Build a request to fetch the vSwitchID from the tags provided
+	describeVSwitchesRequest := vpc.CreateDescribeVSwitchesRequest()
+	describeVSwitchesRequest.Scheme = "https"
+	describeVSwitchesRequest.RegionId = mpc.RegionID
+	if mpc.VpcID != "" {
+		describeVSwitchesRequest.VpcId = mpc.VpcID
 	}
-
-	if vSwitchID == "" {
-		return "", fmt.Errorf("no vswitches were found")
+	describeVSwitchesRequest.Tag = buildDescribeVSwitchesTag(mpc.VSwitch.Tags)
+	describeVSwitchesResponse, err := client.DescribeVSwitches(describeVSwitchesRequest)
+	if err != nil {
+		metrics.RegisterFailedInstanceCreate(&metrics.MachineLabels{
+			Name:      machine.Name,
+			Namespace: machine.Namespace,
+			Reason:    err.Error(),
+		})
+		klog.Errorf("error describing vswitches: %v", err)
+		return "", fmt.Errorf("error describing vswitches: %v", err)
 	}
-
-	return vSwitchID, nil
+	if len(describeVSwitchesResponse.VSwitches.VSwitch) < 1 {
+		klog.Errorf("no vswitches for given tags not found")
+		return "", fmt.Errorf("no vswitches for given tags not found")
```

Comment:
These messages are not parsing for me, what are we trying to say here?

Reply:
This was left over from the original code. I will update. Basically with the provided tags, vpcid, and regionid the query did not return any security groups.